topdown/object.go

```diff
@@ -107,7 +107,36 @@ func builtinObjectFilter(_ BuiltinContext, operands []*ast.Term, iter func(*ast.
 	return iter(ast.NewTerm(r))
 }
 
-func builtinObjectGet(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
+func builtinObjectGet(bctx BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
+	if r, ok := operands[0].Value.(ast.Ref); ok {
```

Comment:
This special path doesn't account for the possibility that operand 2 and 3 are Refs, but operand 1 is not. This is why we need a good interface in `BuiltinContext` to resolve operands that may have not yet been resolved.